[WT-263]  - Change way of reporting files and folders with same name uploaded

CandelR opened the pull request:
- Added logic and modal to handle name collision when uploading or moving elements

aboredvaro committed:
(WIP) — Fixed UI issues and added name collision dialog

aboredvaro committed:
Finished name collision + minor UI fixes

— Fixed styles that where different from Figma design files.
— Fixed HeadlessUI RadioGroup bug.
— Improved accessiblity on multiple modals.

CandelR committed:
Added Rename items modal and logic - WIP

CandelR committed:
Refactor and fix moveDestinationFolderId not reseting correctly

CandelR committed:
Merge branch 'master' into feature/WT-263-change-way-of-reporting-files-with-same-name

CandelR committed:
Merge branch 'master' into origin/feature/WT-263-name-collision-dialog

aboredvaro reviewed (changes requested):
Issues found:
- Must use the correct dialog from [name collision dialog branch](https://github.com/internxt/drive-web/tree/origin/feature/WT-263-name-collision-dialog) to ensure right design implementation and usability standards.
- When user selects "Keep both", everything goes as expected, but an error "Folder not found" pops up.
- When user selects "Replace current item" the folder corrupts, the item that is being replaced is moved to trash (this is ok), but then the new item that is uploaded never gets uploaded (a bunch of errors appear) and from that moment user can't upload anything to that parent folder. [See video for example](https://send.internxt.com/download/0c9cdec7-f21e-4660-958a-6e25f92293c9?code=6b975ecda788889faac28e2d12ecba260417a0f2b0dea3cc76a71302d098a20b).

CandelR committed:
Merge branch 'origin/feature/WT-263-name-collision-dialog' into feature/WT-263-change-way-of-reporting-files-with-same-name

CandelR committed:
Added new name collision dilaog design and refactored logic

CandelR committed:
Merge branch 'master' into feature/WT-263-change-way-of-reporting-files-with-same-name

aboredvaro commented on src/app/drive/components/DriveExplorer/DriveExplorer.tsx:
```diff
@@ -300,6 +353,9 @@ const DriveExplorer = (props: DriveExplorerProps): JSX.Element => {
                       <p className="ml-3">{i18n.get('actions.upload.folder')}</p>
                     </MenuItem>,
+                    <MenuItem onClick={() => dispatch(uiActions.setIsNameCollisionDialogOpen(true))}>
+                      <p>Test name collision</p>
```
Remove this testing menu item
<img width="281" alt="Screenshot 2023-01-18 at 16 57 33" src="https://user-images.githubusercontent.com/34165099/213224533-f1d04c25-fabb-4f12-84e5-c1e61d786446.png">

CandelR committed:
Removed unnecessary code and little refactor

aboredvaro committed:
Various fixes

- Fixed z-index issue where user avatar and task logger showed above the dialogs.
- Fixed modals with wrong styles and bad accessibility practices.
- Added some hard-coded text to the locales file.
